EventFilter/L1TRawToDigi/plugins/implementations_stage2/BMTFUnpackerOutput.cc

```diff
@@ -13,78 +13,76 @@ namespace l1t
+        getBXRange(nBX, firstBX, lastBX);
+        res->setBXRange(firstBX, lastBX);
       }
       else {
-	res->setBXRange(-2, 2);
-	LogDebug("L1T") << "No BXs included in the given Block. Set the BXRange to be (-2,2).";
-	return true;
+        res->setBXRange(-2, 2);
+        LogDebug("L1T") << "No BXs included in the given Block. Set the BXRange to be (-2,2).";
+        return true;
       }
 
       LogDebug("L1T") << "nBX = " << nBX << " firstBX = " << firstBX << " lastBX = " << lastBX;
-			
+
       int processor = block.amc().getBoardID() - 1;
       if ( processor < 0 || processor > 11 )
-	{
-	  edm::LogInfo ("l1t:stage2::BMTFUnpackerOutput::unpack") << "Processor found out of range so it will be calculated by the old way";
-	  if ( block.amc().getAMCNumber()%2 != 0 )
-	    processor = block.amc().getAMCNumber()/2 ;
-	  else
-	    processor = 6 + (block.amc().getAMCNumber()/2 -1);
-	}
+      {
+        edm::LogInfo ("l1t:stage2::BMTFUnpackerOutput::unpack") << "Processor found out of range so it will be calculated by the old way";
+        if ( block.amc().getAMCNumber()%2 != 0 )
+        processor = block.amc().getAMCNumber()/2 ;
+        else
+        processor = 6 + (block.amc().getAMCNumber()/2 -1);
+      }
 
       for (auto bxBlock : bxBlocks) {
-	int ibx = bxBlock.header().getBx();
-
-	for (auto iw = 0; iw < 6; iw+=2) {
-	  uint32_t raw_first = bxBlock.payload()[iw];//payload[ip+(ibx+lastBX)*6];
-	  uint32_t raw_secnd = bxBlock.payload()[iw+1];//payload[ip+(ibx+lastBX)*6];
-	  if ( raw_first == 0 )
-	    {
-	      LogDebug("L1T") << "Raw data is zero";
-	      continue;
-	    }
-					
-	  RegionalMuonCand muCand;
-	  RegionalMuonRawDigiTranslator::fillRegionalMuonCand(muCand, raw_first, raw_secnd, processor, tftype::bmtf);
-
-	  if (muCand.hwQual() == 0)
-	    continue;
-
-	  muCand.setLink(48 + processor);	//the link corresponds to the uGMT input
-	  if (isKalman) {
-	    muCand.setHwPt2((raw_secnd >> 23) & 0xFF);
-	    muCand.setHwDXY((raw_secnd >> 2) & 0x3);
-	    LogDebug("L1T") << "Pt = " << muCand.hwPt() << " eta: " << muCand.hwEta() << " phi: " << muCand.hwPhi() << " diplacedPt = " << muCand.hwPt2();
-	  }
-	  else
-	    LogDebug("L1T") << "Pt = " << muCand.hwPt() << " eta: " << muCand.hwEta() << " phi: " << muCand.hwPhi();
-
-	  res->push_back(ibx, muCand);
-
-	}//for iw
-      }//for ibx
+        int ibx = bxBlock.header().getBx();
+
+        for (auto iw = 0; iw < 6; iw+=2) {
+           uint32_t raw_first = bxBlock.payload()[iw];//payload[ip+(ibx+lastBX)*6];
+           uint32_t raw_secnd = bxBlock.payload()[iw+1];//payload[ip+(ibx+lastBX)*6];
+           if ( raw_first == 0 )
+           {
+             LogDebug("L1T") << "Raw data is zero";
+             continue;
+           }
+
+           RegionalMuonCand muCand;
+           RegionalMuonRawDigiTranslator::fillRegionalMuonCand(muCand, raw_first, raw_secnd, processor, tftype::bmtf, isKalman);
+
+           if (muCand.hwQual() == 0)
```

Comment:
```suggestion
           if (muCand.hwPt() == 0)
```

Reply:
Thanks for this! You could move a slightly modified check a bit higher (before filling the muon) to avoid working on a known empty muon. (See e.g. https://github.com/dinyar/cmssw/blob/ugmt_kalman-adaptions/EventFilter/L1TRawToDigi/plugins/implementations_stage2/RegionalMuonGMTUnpacker.cc#L86) I don't think it would make a significant difference though, let me know if you'd like to change it or if I should just commit your change as it currently is.

Reply:
Hi Dinyar, and sorry for that late reply
lets proceed on committing the code as above, and I can modify it later
what you propose is a good change

Reply:
Ok, sounds good!